diff --git a/cli/cppcheckexecutor.cpp b/cli/cppcheckexecutor.cpp
index 4597affc8..32875ef36 100644
--- a/cli/cppcheckexecutor.cpp
+++ b/cli/cppcheckexecutor.cpp
@@ -833,9 +833,10 @@ int CppCheckExecutor::check_internal(CppCheck& cppcheck, int /*argc*/, const cha
         const std::list<ErrorLogger::ErrorMessage::FileLocation> callstack;
         const std::string msg("Failed to load " + std::string(!std ? "std.cfg" : !posix ? "posix.cfg" : "windows.cfg") + ". Your Cppcheck installation is broken, please re-install.");
 #ifdef FILESDIR
-        const std::string details("The Cppcheck binary was compiled with FILESDIR set to \""
-                                  FILESDIR "\" and will therefore search for "
-                                  "std.cfg in " FILESDIR "/cfg.");
+        std::string filesdir = std::string(CFGDIR).c_str();
+        const std::string details("The Cppcheck binary was compiled with FILESDIR set to \"" +
+                                  filesdir + "\" and will therefore search for "
+                                  "std.cfg in " + filesdir + "/cfg.");
 #else
         const std::string cfgfolder(Path::fromNativeSeparators(Path::getPathFromFilename(argv[0])) + "cfg");
         const std::string details("The Cppcheck binary was compiled without FILESDIR set. Either the "
diff --git a/gui/mainwindow.cpp b/gui/mainwindow.cpp
index 6f2644cc8..88e80c34b 100644
--- a/gui/mainwindow.cpp
+++ b/gui/mainwindow.cpp
@@ -736,7 +736,7 @@ Library::Error MainWindow::loadLibrary(Library *library, const QString &filename
 
 #ifdef FILESDIR
     // Try to load the library from FILESDIR/cfg..
-    const QString filesdir = FILESDIR;
+    const QString filesdir = str::string(FILESDIR).c_str();
     if (!filesdir.isEmpty()) {
         ret = library->load(nullptr, (filesdir+"/cfg/"+filename).toLatin1());
         if (ret.errorcode != Library::ErrorCode::FILE_NOT_FOUND)
diff --git a/gui/projectfiledialog.cpp b/gui/projectfiledialog.cpp
index c9b51b1d4..dfabe8486 100644
--- a/gui/projectfiledialog.cpp
+++ b/gui/projectfiledialog.cpp
@@ -81,7 +81,7 @@ ProjectFileDialog::ProjectFileDialog(ProjectFile *projectFile, QWidget *parent)
     searchPaths << appPath << appPath + "/cfg" << inf.canonicalPath();
 #ifdef FILESDIR
     if (FILESDIR[0])
-        searchPaths << FILESDIR << FILESDIR "/cfg";
+        searchPaths << std::string(FILESDIR).c_str() << std::string(FILESDIR).c_str() "/cfg";
 #endif
     if (!datadir.isEmpty())
         searchPaths << datadir << datadir + "/cfg";
diff --git a/lib/library.cpp b/lib/library.cpp
index 93b94b9a3..bcaa2ae67 100644
--- a/lib/library.cpp
+++ b/lib/library.cpp
@@ -96,7 +96,7 @@ Library::Error Library::load(const char exename[], const char path[])
 
         std::list<std::string> cfgfolders;
 #ifdef FILESDIR
-        cfgfolders.push_back(FILESDIR "/cfg");
+        cfgfolders.push_back(std::string(FILESDIR "/cfg").c_str());
 #endif
         if (exename) {
             const std::string exepath(Path::fromNativeSeparators(Path::getPathFromFilename(exename)));
diff --git a/lib/platform.cpp b/lib/platform.cpp
index d70beb082..3ee7a1ea9 100644
--- a/lib/platform.cpp
+++ b/lib/platform.cpp
@@ -166,7 +166,7 @@ bool cppcheck::Platform::loadPlatformFile(const char exename[], const std::strin
             filenames.push_back(Path::getPathFromFilename(Path::fromNativeSeparators(exename)) + "platforms/" + filename + ".xml");
         }
 #ifdef FILESDIR
-        std::string filesdir = FILESDIR;
+        std::string filesdir = std::string(FILESDIR).c_str();
         if (!filesdir.empty() && filesdir[filesdir.size()-1] != '/')
             filesdir += '/';
         filenames.push_back(filesdir + ("platforms/" + filename));
-- 
2.21.0

